### ISSUE TITLE
test on the MOT17 test set, results NaN.

### ISSUE THREAD

**Issue by yuzhiyiliu** [xingyizhou/CenterTrack#224]
$  python test.py tracking --exp_id mot17_fulltrain_public --dataset mot --dataset_version 17test --pre_hm --ltrb_amodal --track_thresh 0.4 --pre_thresh 0.5 --load_model ../models/mot17_fulltrain_sc.pth --public_det --load_results ../data/mot17/results/test_det.json

|nemot17_half_public |################################| [2651/2652]|Tot: 0:18:20 |ETA: 0:00:01 |tot 0.411s (0.415s) |load 0.000s (0.000s) |pre 0.004s (0.005s) |net 0.405s (0.405s) |dec 0.001s (0.003s) |post 0.000s (0.001s) |merge 0.000s (0.000s) |track 0.000s (0.000s) 
gt_type _val_half
gt_files ['../data/mot17/train/MOT17-02-FRCNN/gt/gt_val_half.txt', '../data/mot17/train/MOT17-05-FRCNN/gt/gt_val_half.txt', '../data/mot17/train/MOT17-04-FRCNN/gt/gt_val_half.txt', '../data/mot17/train/MOT17-13-FRCNN/gt/gt_val_half.txt', '../data/mot17/train/MOT17-11-FRCNN/gt/gt_val_half.txt', '../data/mot17/train/MOT17-10-FRCNN/gt/gt_val_half.txt', '../data/mot17/train/MOT17-09-FRCNN/gt/gt_val_half.txt']
02:06:40 INFO - Found 7 groundtruths and 7 test files.
02:06:40 INFO - Available LAP solvers ['scipy']
02:06:40 INFO - Default LAP solver 'scipy'
02:06:40 INFO - Loading files.
02:06:41 INFO - Comparing MOT17-02-FRCNN...
02:06:42 INFO - Comparing MOT17-05-FRCNN...
02:06:42 INFO - Comparing MOT17-11-FRCNN...
02:06:42 INFO - Comparing MOT17-09-FRCNN...
02:06:42 INFO - Comparing MOT17-04-FRCNN...
02:06:43 INFO - Comparing MOT17-13-FRCNN...
02:06:43 INFO - Comparing MOT17-10-FRCNN...
02:06:43 INFO - Running metrics
02:06:43 INFO - partials: 0.503 seconds.
02:06:44 INFO - mergeOverall: 0.513 seconds.
               Rcll Prcn  GT   MT   PT     ML   FP     FN  IDs   FM MOTA MOTP num_objects
MOT17-02-FRCNN 0.0% nan%  53 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  nan        9880
MOT17-05-FRCNN 0.0% nan%  71 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  nan        3357
MOT17-11-FRCNN 0.0% nan%  44 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  nan        4517
MOT17-09-FRCNN 0.0% nan%  22 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  nan        2879
MOT17-04-FRCNN 0.0% nan%  69 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  nan       24178
MOT17-13-FRCNN 0.0% nan%  44 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  nan        3156
MOT17-10-FRCNN 0.0% nan%  36 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  nan        5923
OVERALL        0.0% nan% 339 0.0% 0.0% 100.0% 0.0% 100.0% 0.0% 0.0% 0.0%  nan       53890
02:06:44 INFO - partials: 0.811 seconds.
02:06:44 INFO - mergeOverall: 0.824 seconds.
               IDF1  IDP  IDR Rcll Prcn  GT MT PT  ML FP    FN IDs  FM MOTA MOTP IDt IDa IDm num_objects
MOT17-02-FRCNN 0.0% nan% 0.0% 0.0% nan%  53  0  0  53  0  9880   0   0 0.0%  nan   0   0   0        9880
MOT17-05-FRCNN 0.0% nan% 0.0% 0.0% nan%  71  0  0  71  0  3357   0   0 0.0%  nan   0   0   0        3357
MOT17-11-FRCNN 0.0% nan% 0.0% 0.0% nan%  44  0  0  44  0  4517   0   0 0.0%  nan   0   0   0        4517
MOT17-09-FRCNN 0.0% nan% 0.0% 0.0% nan%  22  0  0  22  0  2879   0   0 0.0%  nan   0   0   0        2879
MOT17-04-FRCNN 0.0% nan% 0.0% 0.0% nan%  69  0  0  69  0 24178   0   0 0.0%  nan   0   0   0       24178
MOT17-13-FRCNN 0.0% nan% 0.0% 0.0% nan%  44  0  0  44  0  3156   0   0 0.0%  nan   0   0   0        3156
MOT17-10-FRCNN 0.0% nan% 0.0% 0.0% nan%  36  0  0  36  0  5923   0   0 0.0%  nan   0   0   0        5923
OVERALL        0.0% nan% 0.0% 0.0% nan% 339  0  0 339  0 53890   0   0 0.0%  nan   0   0   0       53890
02:06:44 INFO - Completed

I'm sure the env set up correctly, because  `test the tracking performance on MOT17 with our pretrained model` is normal, got 66.1%.

**Comment by Hidehidden** [comment]
Excuse me.Have you found the solution?Thank you in advance.

**Comment by ferumchrome** [comment]
@xingyizhou 

Hello, I have the same problem while using -public_det

**Comment by LesterLian** [comment]
Check the JSON file in mot17/results and see if it contains any bbox. If not, maybe my changes to the result converting script in #232 can help.

**Comment by xjtulyz** [comment]
I have the same problem with you. Have you solved the problem yet?  Thanks

**Comment by LesterLian** [comment]
@Li1zz 
Have you tried my fix? If it doesn't help, you are mostly on your own. There're incomplete code here and there, yet I don't think the author actively maintains this project.

**Comment by Li1zz** [comment]
@LesterLian 
Hello, I saw the solution you provide , but i dont know how to implement it at that time. I guess it is to replace convert_mot_det_to_results.py  file with the fix you provided, and then run datasetup again, is it right?

**Comment by Li1zz** [comment]
@LesterLian 
Running is successful after I tried your solution to this.   Thank you so much !
